packages/astro/src/content/loaders/glob.ts

```diff
@@ -63,41 +62,87 @@ export function glob(globOptions: GlobOptions): Loader {
 
 	const fileToIdMap = new Map<string, string>();
 
-	async function syncData(
-		entry: string,
-		base: URL,
-		{ logger, parseData, store }: LoaderContext,
-		entryType?: ContentEntryType
-	) {
-		if (!entryType) {
-			logger.warn(`No entry type found for ${entry}`);
-			return;
-		}
-		const fileUrl = new URL(entry, base);
-		const { body, data } = await entryType.getEntryInfo({
-			contents: await fs.readFile(fileUrl, 'utf-8'),
-			fileUrl,
-		});
-
-		const id = generateId({ entry, base, data });
-
-		const filePath = fileURLToPath(fileUrl);
-
-		const parsedData = await parseData({
-			id,
-			data,
-			filePath,
-		});
-
-		store.set(id, parsedData, body, filePath);
-
-		fileToIdMap.set(filePath, id);
-	}
-
 	return {
 		name: 'glob-loader',
-		load: async (options) => {
-			const { settings, logger, watcher } = options;
+		load: async ({ settings, logger, watcher, parseData, store, generateDigest }) => {
+			const renderFunctionByContentType = new WeakMap<ContentEntryType, RenderFunction>();
+
+			const untouchedEntries = new Set(store.keys());
+
+			async function syncData(entry: string, base: URL, entryType?: ContentEntryType) {
+				if (!entryType) {
+					logger.warn(`No entry type found for ${entry}`);
+					return;
+				}
+				const fileUrl = new URL(entry, base);
+				const contents = await fs.readFile(fileUrl, 'utf-8');
```

Comment:
Possible runtime error to catch (file doesn't exist, or no rights to read the file)